test/distributed/test_fake_pg.py

```diff
@@ -168,6 +175,73 @@ def test_recv(self):
         dist.recv(output, 1)
         self.assertEqual(tuple(output.shape), (3, 3))
 
+    @unittest.skipIf(not HAS_CUDA or not enable_2d_with_fsdp(), "No CUDA or TP+FSDP")
+    def test_fsdp_tp_sp_fake_e2e(self):
+        world_size = 4
+        tp_size = 2
+        parallel_style = SequenceParallel()
+
+        store = dist.HashStore()
+        dist.init_process_group(backend="fake", rank=0, world_size=world_size, store=store)
+
+        device_mesh = DeviceMesh(
+            "cuda", torch.arange(0, world_size).view(-1, tp_size)
+        )
+
+        my_module = nn.Sequential(
+            nn.Linear(10, 32),
+            nn.ReLU(),
+            nn.Linear(32, 5),
+        ).cuda(dist.get_rank())
```

Comment:
Can we reuse the module defined here: https://github.com/pytorch/pytorch/blob/9c56ca80f36f98cb01190eb63053b07c15e95038/torch/testing/_internal/distributed/_tensor/common_dtensor.py#L53. Or if you don't want to use that one, can we make this more reusable so that we avoid duplicate code? Thanks!